python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -270,6 +270,57 @@ def compute_child_subset(
                 value=_ValidatedEntitySubsetValue(child_partitions_subset),
             )
 
+    def compute_mapped_subset(
+        self, to_key: T_EntityKey, from_subset: EntitySubset[AssetKey]
+    ) -> EntitySubset[T_EntityKey]:
+        from_key = from_subset.key
+        from_partitions_def = self.asset_graph.get(from_key).partitions_def
+        to_partitions_def = self.asset_graph.get(to_key).partitions_def
+
+        partition_mapping = self.asset_graph.get_partition_mapping(from_key, to_key)
+
+        if from_partitions_def is None or to_partitions_def is None:
+            return (
+                self.get_empty_subset(key=to_key)
+                if from_subset.is_empty
+                else self.get_full_subset(key=to_key)
+            )
+
+        if from_key in self.asset_graph.get(to_key).parent_entity_keys:
+            to_partitions_subset = partition_mapping.get_downstream_partitions_for_partitions(
+                upstream_partitions_subset=from_subset.get_internal_subset_value(),
+                upstream_partitions_def=from_partitions_def,
+                downstream_partitions_def=to_partitions_def,
+                dynamic_partitions_store=self._queryer,
+                current_time=self.effective_dt,
+            )
+        elif from_key in self.asset_graph.get(to_key).child_entity_keys:
+            to_partitions_subset = (
+                partition_mapping.get_upstream_mapped_partitions_result_for_partitions(
+                    downstream_partitions_subset=from_subset.get_internal_subset_value(),
+                    downstream_partitions_def=from_partitions_def,
+                    upstream_partitions_def=to_partitions_def,
+                    dynamic_partitions_store=self._queryer,
+                    current_time=self.effective_dt,
+                ).partitions_subset
+            )
+        else:
+            to_partitions_subset = (
+                partition_mapping.get_upstream_mapped_partitions_result_for_partitions(
+                    downstream_partitions_subset=None,
```

Comment:
While the parameter is called "downstream_partitions_subset", we should still be passing in from_subset.get_internal_subset_value() (and the partitions_def) here. Otherwise, we'll always be treating the from_key as unpartitioned.

This makes the last two cases identical, so they can just be grouped under a single `else`.